crates/bevy_gizmos/src/circles.rs

```diff
@@ -4,20 +4,102 @@
 //! and assorted support items.
 
 use crate::prelude::{GizmoConfigGroup, Gizmos};
+use bevy_math::Mat2;
 use bevy_math::{primitives::Direction3d, Quat, Vec2, Vec3};
 use bevy_render::color::Color;
 use std::f32::consts::TAU;
 
 pub(crate) const DEFAULT_CIRCLE_SEGMENTS: usize = 32;
 
-fn circle_inner(radius: f32, segments: usize) -> impl Iterator<Item = Vec2> {
+fn ellipse_inner(half_width: f32, half_height: f32, segments: usize) -> impl Iterator<Item = Vec2> {
     (0..segments + 1).map(move |i| {
         let angle = i as f32 * TAU / segments as f32;
-        Vec2::from(angle.sin_cos()) * radius
+        let (x, y) = angle.sin_cos();
+        Vec2::new(x * half_width, y * half_height)
     })
 }
 
 impl<'w, 's, T: GizmoConfigGroup> Gizmos<'w, 's, T> {
+    /// Draw a ellipse in 3D at `position` with the flat side facing `normal`.
+    ///
+    /// This should be called for each frame the ellipse needs to be rendered.
+    ///
+    /// # Example
+    /// ```
+    /// # use bevy_gizmos::prelude::*;
+    /// # use bevy_render::prelude::*;
+    /// # use bevy_math::prelude::*;
+    /// fn system(mut gizmos: Gizmos) {
+    ///     gizmos.ellipse(Vec3::ZERO, Vec3::Z, 1., 2., Color::GREEN);
+    ///
+    ///     // Ellipses have 32 line-segments by default.
+    ///     // You may want to increase this for larger ellipses.
+    ///     gizmos
+    ///         .ellipse(Vec3::ZERO, Quat::IDENTITY, Vec3::Z, 5., 1., Color::RED)
+    ///         .segments(64);
+    /// }
+    /// # bevy_ecs::system::assert_is_system(system);
+    /// ```
+    #[inline]
+    pub fn ellipse(
+        &mut self,
+        position: Vec3,
+        rotation: Quat,
+        half_width: f32,
+        half_height: f32,
+        color: Color,
+    ) -> EllipseBuilder<'_, 'w, 's, T> {
+        EllipseBuilder {
+            gizmos: self,
+            position,
+            rotation,
+            half_width,
+            half_height,
+            color,
+            segments: DEFAULT_CIRCLE_SEGMENTS,
+        }
+    }
+
+    /// Draw a ellipse in 2D.
+    ///
+    /// This should be called for each frame the ellipse needs to be rendered.
+    ///
+    /// # Example
+    /// ```
+    /// # use bevy_gizmos::prelude::*;
+    /// # use bevy_render::prelude::*;
+    /// # use bevy_math::prelude::*;
+    /// fn system(mut gizmos: Gizmos) {
+    ///     gizmos.ellipse_2d(Vec2::ZERO, 1., 2., Color::GREEN);
+    ///
+    ///     // Ellipses have 32 line-segments by default.
+    ///     // You may want to increase this for larger ellipses.
+    ///     gizmos
+    ///         .ellipse_2d(Vec2::ZERO, Mat2::IDENTITY, 5., 1., Color::RED)
+    ///         .segments(64);
+    /// }
+    /// # bevy_ecs::system::assert_is_system(system);
+    /// ```
+    #[inline]
+    pub fn ellipse_2d(
+        &mut self,
+        position: Vec2,
+        rotation: Mat2,
```

Comment:
No strong preference here, but `Mat2` might not be as user-friendly as just passing an `f32` angle in radians (like the bounding impls currently). If you don't know `Mat2::from_angle` exists, it'll probably be confusing to use.

My plan is to eventually add a `Rotation2d` type which would store a unit complex number (sine and cosine of rotation angle) and be efficient like `Mat2`. It'd implement `From<f32>` to create it from radians, so e.g. methods like this could take an `impl Into<Rotation2d>` and accept just radians (or a `Mat2` if desired) which are automatically converted to the rotation type